### PR TITLE
fix: workaround to avoid eslint-disable

### PR DESCRIPTION
**Pull request by nensidosari**
## Changes
- added tabIndex={0} and role="link" in order to avoid disabling eslint for `jsx-a11y/no-static-element-interactions`
- added empty <span/> tag in order to avoid disabling eslint for `jsx-a11y/anchor-has-content`
- added onKeyDown event to make the link be clicked and work with Spacebar as well (as it was working as a button) in order to avoid disabling eslint for `jsx-a11y/click-events-have-key-events`

### Describe what this PR does
- Cleans up code from eslint-disbales

## Events

Did you introduce any new tracking events? No

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension? No
- [ ] Does this not break anything in companion? No

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices? No
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1891 #done

### CODE REVIEW

File: packages/shared/src/components/notifications/NotificationItem.tsx [hunk 1]
```diff
@@ -63,14 +69,24 @@ function NotificationItem({
...
+            role="link"
             aria-label="Open notification"
             className="absolute inset-0 z-0"
             onClick={onClick}
             title="Open notification"
             data-testid="openNotification"
-          />
+            tabIndex={0}
+            onKeyDown={(e) => {
+              if (e.code === 'Space') {
```
Comment (DragosIorgulescu) [inline packages/shared/src/components/notifications/NotificationItem.tsx:81]: should we add this to the elements enum? https://github.com/dailydotdev/apps/blob/7c546fbf1006a430b9c54586ffd1094a30a2d486/packages/shared/src/lib/element.ts#L3-L5

**Review by DragosIorgulescu — COMMENTED** [review]
just 1 small question, otherwise looks good 🙏

File: packages/shared/src/components/notifications/NotificationItem.tsx (hunk 1 above)
Comment (nensidosari) [reply packages/shared/src/components/notifications/NotificationItem.tsx]: Yeah sure, done :D

**Review by sshanzel — APPROVED** [review]
Looks so great without the disables 😍 

Thank you for the fix and the references on how to work it around! 🙇 🙏

**Review by DragosIorgulescu — APPROVED** [review]
💯

**Pull request merged**